core/optaplanner-core-impl/src/main/java/org/optaplanner/core/impl/domain/solution/cloner/BooleanFieldCloner.java

```diff
@@ -0,0 +1,43 @@
+package org.optaplanner.core.impl.domain.solution.cloner;
+
+import java.lang.reflect.Field;
+import java.util.Optional;
+
+final class BooleanFieldCloner<C> implements FieldCloner<C> {
+
+    private static final FieldCloner INSTANCE = new BooleanFieldCloner();
+
+    public static <C> FieldCloner<C> getInstance() {
+        return INSTANCE;
+    }
+
+    @Override
+    public Optional<Unprocessed> clone(DeepCloningUtils deepCloningUtils, Field field, Class<? extends C> instanceClass,
+            C original, C clone) {
+        boolean originalValue = getFieldValue(original, field);
+        setFieldValue(clone, field, originalValue);
+        return Optional.empty();
+    }
+
+    private static boolean getFieldValue(Object bean, Field field) {
+        try {
+            return field.getBoolean(bean);
+        } catch (IllegalAccessException e) {
+            FieldCloner.failOnRead(bean, field, e);
+            return false;
```

Comment:
I find this pattern odd;  `FieldCloner.failOnRead` will ALWAYS throw an exception, but you still return a value (despite it will never be reached). Maybe make `FieldCloner.failOnRead(bean, field, e)` return an exception, and then throw it?

```suggestion
            throw FieldCloner.failOnRead(bean, field, e);
```